### PR TITLE
test(dynamite): test authentication in e2e

### PR DESCRIPTION
**Pull request by Leptopoda**
Signed-off-by: Nikolas Rimikis <leptopoda@users.noreply.github.com>

### CODE REVIEW

File: packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      when(response.statusCode).thenReturn(200);
+
+      request = MockHttpClientRequest();
+      when(request.headers).thenReturn(headers);
+      when(request.close()).thenAnswer((_) async => response);
+      return Future.value(request);
+    });
+  });
+
+  test('noAuthentication', () async {
```
Comment (provokateurin) [inline packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart:53]:
```suggestion
  test('No authentication', () async {
```

File: packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    expect(captured, equals(['Accept', 'application/json']));
+
+    // multiple registered authentication
+    client = $Client(uri, authentications: const [basicAuth, bearerAuth]);
+    await client.noAuthentication();
+    captured = verify(headers.add(captureAny, captureAny)).captured;
+    expect(captured, equals(['Accept', 'application/json']));
+  });
+
+  test('basicAuthentication', () async {
```
Comment (provokateurin) [inline packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart:79]:
```suggestion
  test('Basic authentication', () async {
```

File: packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      equals([
+        'Accept',
+        'application/json',
+        'Authorization',
+        'Basic YmVhcmVyLXVzZXJuYW1lOmJlYXJlci1wYXNzd29yZA==',
+      ]),
+    );
+  });
+
+  test('bearerAuthentication', () async {
```
Comment (provokateurin) [inline packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart:117]:
```suggestion
  test('Bearer authentication', () async {
```

File: packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      equals([
+        'Accept',
+        'application/json',
+        'Authorization',
+        'Bearer bearer-token',
+      ]),
+    );
+  });
+
+  test('multipleAuthentications', () async {
```
Comment (provokateurin) [inline packages/dynamite/dynamite_end_to_end_test/test/authentication_test.dart:155]:
```suggestion
  test('Multiple authentications', () async {
```

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Review by provokateurin — APPROVED** [review]

File: packages/dynamite/dynamite_end_to_end_test/lib/authentication.openapi.json [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    "openapi": "3.1.0",
+    "info": {
+        "title": "comments",
+        "version": "0.0.1",
+        "description": "Files app plugin to add comments to files",
+        "license": {
+            "name": "agpl",
+            "identifier": "AGPL-3.0-only"
+        }
+    },
```
Comment (Leptopoda) [inline packages/dynamite/dynamite_end_to_end_test/lib/authentication.openapi.json:11]: Copy pasta

**Pull request merged**